godot-core/src/registry.rs

```diff
@@ -200,19 +244,58 @@ pub fn auto_register_classes(init_level: InitLevel) {
         fill_class_info(elem.component.clone(), class_info);
     });
 
-    //out!("Class-map: {map:#?}");
+    let mut loaded_classes = get_loaded_classes_with_mutex();
+    let loaded_classes = loaded_classes.get_or_insert_with(Default::default);
 
     for info in map.into_values() {
         out!(
             "Register class:   {} at level `{init_level:?}`",
             info.class_name
         );
+        let class_name = info.class_name;
+        loaded_classes
+            .entry(init_level)
+            .or_default()
+            .push(info.class_name);
         register_class_raw(info);
+
+        godot_print!("Class {} loaded", class_name);
     }
 
     out!("All classes for level `{init_level:?}` auto-registered.");
 }
 
+pub fn unregister_classes(init_level: InitLevel) {
+    let mut loaded_classes = get_loaded_classes_with_mutex();
+    let loaded_classes = loaded_classes.get_or_insert_with(Default::default);
+    let loaded_classes = loaded_classes.remove(&init_level).unwrap_or_default();
```

Comment:
Naming both these things `loaded_classes` is a bit confusing. As above, I'd name the map `loaded_class_map`.